### PR TITLE
Switch to ament_cmake_vendor_package

### PR DESCRIPTION
**Pull request by cottsay**
Part of ament/ament_cmake#429

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18830/)
* Linux [Build Status](https://ci.ros2.org/job/ci_linux/18831/) (FORCE_BUILD_VENDOR_PKG)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13359/)
* Linux-rhel [Build Status](https://ci.ros2.org/job/ci_linux-rhel/421/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19575/)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,87 +1,23 @@
...
-  list(APPEND extra_cmake_args "-DSPDLOG_BUILD_EXAMPLE=OFF")
-  if(NOT WIN32 AND BUILD_SHARED_LIBS)
-    list(APPEND extra_cmake_args "-DSPDLOG_BUILD_SHARED=ON")
-  endif()
+find_package(spdlog 1.5.0 QUIET)
 
-  include(ExternalProject)
+# While most CMake projects build shared libraries by default, it appears that
+# spdlog needs to be told to do so explicitly.
+set(BUILD_SHARED_LIBS TRUE CACHE BOOL "Build shared libraries")
```
Comment (clalancette) [inline CMakeLists.txt:11]: Out of curiousity, why is this set here rather than being passed through to `ament_vendor` as a `CMAKE_ARGS`?

**Review by clalancette — COMMENTED** [review]
I have a question inline, but this otherwise looks great to me.  Love the reduction of the boilerplate code here.

File: CMakeLists.txt (hunk 1 above)
Comment (cottsay) [reply CMakeLists.txt:11]: `BUILD_SHARED_LIBS` is one of the selected CMake variables that get implicitly passed to the vendor project if defined. It was less code to treat it as a project-local variable than it was to add conditional logic for adding it to the explicit `CMAKE_ARGS` to the `ament_vendor` call.

**Comment by quarkytale** [comment]
This is the first time I saw a green Windows build!

**Review by quarkytale — APPROVED** [review]

**Pull request merged**